Issue in compilation error msg order

**Description:**
The following error occurs due to concurrency issue in the semantic analyzer.

```
[ERROR] Tests run: 4095, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 268.458 s <<< FAILURE! - in TestSuite [ERROR] testSemanticErrors(org.ballerinalang.test.statements.foreach.ForeachNegativeTests) Time elapsed: 0.137 s <<< FAILURE! java.lang.AssertionError: incorrect error message: expected [redeclared symbol 's'] but found [redeclared symbol 'i'] at org.ballerinalang.test.statements.foreach.ForeachNegativeTests.testSemanticErrors(ForeachNegativeTests.java:42)
```

--- a/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java
+++ b/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java
@@ -1089,13 +1089,12 @@ public class BLangPackageBuilder {
         switch (variable.getKind()) {
             case TUPLE_VARIABLE:
                 // If the variable is a tuple variable, we need to set the final flag to the all member variables.
-                ((BLangTupleVariable) variable).memberVariables.parallelStream()
-                        .forEach(this::markVariableAsFinal);
+                ((BLangTupleVariable) variable).memberVariables.forEach(this::markVariableAsFinal);
                 break;
             case RECORD_VARIABLE:
                 // If the variable is a record variable, we need to set the final flag to the all the variables in
                 // the record.
-                ((BLangRecordVariable) variable).variableList.parallelStream()
+                ((BLangRecordVariable) variable).variableList.stream()
                         .map(BLangRecordVariableKeyValue::getValue)
                         .forEach(this::markVariableAsFinal);
                 break;

--- a/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java
+++ b/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java
@@ -644,12 +644,12 @@ public class SemanticAnalyzer extends BLangNodeVisitor {
                 symbolEnter.defineVarSymbol(variable.pos, variable.flagSet, variable.type, name, blockEnv);
                 break;
             case TUPLE_VARIABLE:
-                ((BLangTupleVariable) variable).memberVariables.parallelStream()
-                        .forEach(memberVariable -> recursivelyDefineVariables(memberVariable, blockEnv));
+                ((BLangTupleVariable) variable).memberVariables.forEach(memberVariable ->
+                        recursivelyDefineVariables(memberVariable, blockEnv));
                 break;
             case RECORD_VARIABLE:
-                ((BLangRecordVariable) variable).variableList.parallelStream()
-                        .forEach(value ->recursivelyDefineVariables(value.valueBindingPattern, blockEnv));
+                ((BLangRecordVariable) variable).variableList.forEach(value ->
+                        recursivelyDefineVariables(value.valueBindingPattern, blockEnv));
                 break;
         }
     }

--- a/tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/statements/foreach/ForeachNegativeTests.java
+++ b/tests/ballerina-unit-test/src/test/java/org/ballerinalang/test/statements/foreach/ForeachNegativeTests.java
@@ -39,8 +39,8 @@ public class ForeachNegativeTests {
                 "'string' in type definition", 4, 17);
         BAssertUtil.validateError(compile, index++, "invalid tuple variable; expecting a tuple type but found " +
                 "'string' in type definition", 13, 17);
-        BAssertUtil.validateError(compile, index++, "redeclared symbol 's'", 13, 21);
         BAssertUtil.validateError(compile, index++, "redeclared symbol 'i'", 13, 18);
+        BAssertUtil.validateError(compile, index++, "redeclared symbol 's'", 13, 21);
         BAssertUtil.validateError(compile, index++, "invalid tuple variable; expecting a tuple type but found " +
                 "'string' in type definition", 20, 17);
         BAssertUtil.validateError(compile, index++, "undefined symbol 'i'", 23, 16);

